### ISSUE TITLE
Errors cam be reported when uptime OID is not present

### ISSUE THREAD

**Issue by DmatteoCM** [Cacti/spine#315]
**Describe the bug**
Hello everyone, I have the spine logs full of these errors:

```
2023/09/07 16:15:06 - SPINE: Poller[Main Poller] PID[33406] PT[139880030050048] ERROR: No such Instance for oid '.1.3.6.1.6.3.10.2.1.3.0' for Device[VDSL - Enomet] with Status[1]
2023/09/07 16:15:05 - SPINE: Poller[Main Poller] PID[33406] PT[139880030050048] ERROR: No such Instance for oid '.1.3.6.1.6.3.10.2.1.3.0' for Device[VDSL - Enomet] with Status[1]
```

I have about 300 devices and they all show this error. How can I fix it?

thanks

**To Reproduce**

the spine log:
```
#./spine -R -V 5 -f 294 -l 294
SPINE: Using spine config file [../etc/spine.conf]
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065406259392] DEBUG: The path_php_server variable is /cacti/script_server.php
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065406259392] DEBUG: The path_cactilog variable is /var/log/cacti/cacti.log
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065406259392] DEBUG: The version variable is 10.10.2-MariaDB-1:10.10.2+maria~ubu2204
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065406259392] DEBUG: The log_destination variable is 1 (FILE)
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065406259392] DEBUG: The path_php variable is /usr/bin/php
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065406259392] DEBUG: The availability_method variable is 1
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065406259392] DEBUG: The ping_recovery_count variable is 3
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065406259392] DEBUG: The ping_failure_count variable is 2
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065406259392] DEBUG: The ping_method variable is 1
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065406259392] DEBUG: The ping_retries variable is 1
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065406259392] DEBUG: The ping_timeout variable is 400
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065406259392] DEBUG: The snmp_retries variable is 3
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065406259392] DEBUG: The log_perror variable is 0
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065406259392] DEBUG: The log_pwarn variable is 0
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065406259392] DEBUG: The boost_redirect variable is 1
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065406259392] DEBUG: The boost_rrd_update_enable variable is 1
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065406259392] DEBUG: The log_pstats variable is 0
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065406259392] DEBUG: The threads variable is 4
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065406259392] DEBUG: The polling interval is 300 seconds
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065406259392] DEBUG: The number of concurrent processes is 2
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065406259392] DEBUG: The script timeout is 25
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065406259392] DEBUG: The selective_device_debug variable is 15,190,210
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065406259392] DEBUG: The spine_log_level variable is 0
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065406259392] DEBUG: The number of php script servers to run is 10
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065406259392] DEBUG: The number of active data source profiles is 1
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065406259392] DEBUG: The number of snmp ports on the system is 87
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065406259392] DEBUG: StartDevice='294', EndDevice='294', TotalPHPScripts='0'
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065406259392] DEBUG: The PHP Script Server is Not Required
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065406259392] DEBUG: The Maximum SNMP OID Get Size is 10
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065406259392] DEBUG: Selective Debug Devices 15,190,210
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065406259392] DEBUG: Total Connections made 1
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065406259392] DEBUG: Creating Local Connection Pool of 4 threads.
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065406259392] DEBUG: Creating Local Connection 0.
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065406259392] DEBUG: Total Connections made 2
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065406259392] DEBUG: Creating Local Connection 1.
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065406259392] DEBUG: Total Connections made 3
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065406259392] DEBUG: Creating Local Connection 2.
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065406259392] DEBUG: Total Connections made 4
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065406259392] DEBUG: Creating Local Connection 3.
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065406259392] DEBUG: Total Connections made 5
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065406259392] DEBUG: Version 1.2.25 starting
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065406259392] DEBUG: MySQL is Thread Safe!
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065406259392] DEBUG: Spine running as 0 UID, 0 EUID
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065406259392] DEBUG: Spine is running as root.
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065406259392] DEBUG: Spine has got ICMP
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065406259392] DEBUG: Initializing Net-SNMP API
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065406259392] DEBUG: Issues with SNMP Header Version information, assuming old version of Net-SNMP.
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065406259392] DEBUG: Initializing PHP Script Server(s)
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065406259392] DEBUG: Initial Value of Available Threads is 4 (0 outstanding)
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065406259392] DEBUG: Valid Thread to be Created (140065373583104)
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065406259392] DEBUG: Available Threads is 3 (1 outstanding)
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065373583104] DEBUG: Device[0] HT[1] In Poller, About to Start Polling
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065373583104] DEBUG: Traversing Local Connection Pool for free connection.
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065373583104] DEBUG: Checking Local Pool ID 0.
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065373583104] DEBUG: Allocating Local Pool ID 0.
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065406259392] DEBUG: Valid Thread to be Created (140065365190400)
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065406259392] DEBUG: Available Threads is 2 (2 outstanding)
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065406259392] WARNING: Device[294] polling sleeping while waiting for 2 Threads to End
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065373583104] Device[0] HT[1] Total Time: 0.0021 Seconds
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065365190400] DEBUG: Device[294] HT[1] In Poller, About to Start Polling
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065365190400] DEBUG: Traversing Local Connection Pool for free connection.
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065365190400] DEBUG: Checking Local Pool ID 0.
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065365190400] DEBUG: Checking Local Pool ID 1.
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065365190400] DEBUG: Allocating Local Pool ID 1.
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065373583104] DEBUG: Freeing Local Pool ID 0
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065373583104] DEBUG: Device[0] HT[1] DEBUG: HOST COMPLETE: About to Exit Device Polling Thread Function
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065365190400] get_namebyhost(XX.XX.XX.XX) - Allocating name_t
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065365190400] get_namebyhost(XX.XX.XX.XX) - No delimiter, assume full hostname
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065365190400] Device[294] INFO: SNMP Device 'XX.XX.XX.XX:161' has a timeout of 500000 (500), with 3 retries
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065365190400] Device[294] IPv4 address XX.XX.XX.XX (XX.XX.XX.XX)
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065365190400] Device[294] DEBUG: Entering ICMP Ping
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065406259392] WARNING: Device[294] polling sleeping while waiting for 1 Threads to End
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065406259392] WARNING: Device[294] polling sleeping while waiting for 1 Threads to End
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065365190400] Device[294] DEBUG: Attempting to ping XX.XX.XX.XX, seq 0 (Retry 0 of 1)
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065365190400] Device[294] DEBUG: ICMP Device Alive, Try Count:1, Time:80.1961 ms
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065406259392] WARNING: Device[294] polling sleeping while waiting for 1 Threads to End
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065406259392] WARNING: Device[294] polling sleeping while waiting for 1 Threads to End
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065365190400] Device[294] DEBUG: Entering SNMP Ping
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065365190400] Device[294] PING Result: ICMP: Device is Alive
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065365190400] Device[294] SNMP Result: Device responded to SNMP
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065365190400] Device[294] Checking for System Information Update
2023/09/07 16:19:30 - SPINE: Poller[1] PID[35241] PT[140065365190400] Device[294] Updating Full System Information Table
2023/09/07 16:19:31 - SPINE: Poller[1] PID[35241] PT[140065365190400] ERROR: No such Instance for oid '.1.3.6.1.6.3.10.2.1.3.0' for Device[294] with Status[1]
2023/09/07 16:19:31 - SPINE: Poller[1] PID[35241] PT[140065365190400] DEBUG: Device[294] HT[1] RECACHE: Processing 1 items in the auto reindex cache for 'XX.XX.XX.XX'
2023/09/07 16:19:31 - SPINE: Poller[1] PID[35241] PT[140065365190400] Device[294] HT[1] DQ[1] Legacy Uptime Result: 3153400, Is Numeric: 1
2023/09/07 16:19:31 - SPINE: Poller[1] PID[35241] PT[140065365190400] ERROR: No such Instance for oid '.1.3.6.1.6.3.10.2.1.3.0' for Device[294] with Status[1]
2023/09/07 16:19:31 - SPINE: Poller[1] PID[35241] PT[140065365190400] Device[294] HT[1] DQ[1] Extended Uptime Result: 3153400, Is Numeric: 1
2023/09/07 16:19:31 - SPINE: Poller[1] PID[35241] PT[140065365190400] Device[294] HT[1] DQ[1] RECACHE OID: .1.3.6.1.2.1.1.3.0, (assert: 3126500 < output: 3153400)
2023/09/07 16:19:31 - SPINE: Poller[1] PID[35241] PT[140065365190400] Device[294] HT[1] NOTE: There are '11' Polling Items for this Device
2023/09/07 16:19:31 - SPINE: Poller[1] PID[35241] PT[140065365190400] The executable is '/usr/bin/perl' in 'perl /cacti/scripts/ping.pl 'XX.XX.XX.XX''
2023/09/07 16:19:31 - SPINE: Poller[1] PID[35241] PT[140065365190400] DEBUG: Device[294] DEBUG: The NIFTY POPEN returned the following File Descriptor 9
perl: warning: Setting locale failed.
perl: warning: Please check that your locale settings:
        LANGUAGE = (unset),
        LC_ALL = (unset),
        LANG = "en_US.UTF-8"
    are supported and installed on your system.
perl: warning: Falling back to the standard locale ("C").
2023/09/07 16:19:31 - SPINE: Poller[1] PID[35241] PT[140065406259392] WARNING: Device[294] polling sleeping while waiting for 1 Threads to End
2023/09/07 16:19:31 - SPINE: Poller[1] PID[35241] PT[140065365190400] Device[294] HT[1] DS[855] TT[63.85] SCRIPT: perl /cacti/scripts/ping.pl 'XX.XX.XX.XX', output: 38.6
2023/09/07 16:19:31 - SPINE: Poller[1] PID[35241] PT[140065365190400] Device[294] INFO: SNMP Device 'XX.XX.XX.XX:161' has a timeout of 500000 (500), with 3 retries
2023/09/07 16:19:31 - SPINE: Poller[1] PID[35241] PT[140065365190400] Device[294] HT[1] DS[856] TT[38.97] SNMP: v2: XX.XX.XX.XX, dsname: traffic_in, oid: .1.3.6.1.2.1.2.2.1.10.1, value: 1369189469
2023/09/07 16:19:31 - SPINE: Poller[1] PID[35241] PT[140065365190400] Device[294] HT[1] DS[856] TT[41.15] SNMP: v2: XX.XX.XX.XX, dsname: traffic_out, oid: .1.3.6.1.2.1.2.2.1.16.1, value: 3455327763
2023/09/07 16:19:31 - SPINE: Poller[1] PID[35241] PT[140065365190400] Device[294] HT[1] DS[857] TT[42.31] SNMP: v2: XX.XX.XX.XX, dsname: traffic_in, oid: .1.3.6.1.2.1.2.2.1.10.4, value: 939366464
2023/09/07 16:19:31 - SPINE: Poller[1] PID[35241] PT[140065365190400] Device[294] HT[1] DS[857] TT[43.17] SNMP: v2: XX.XX.XX.XX, dsname: traffic_out, oid: .1.3.6.1.2.1.2.2.1.16.4, value: 316100995
2023/09/07 16:19:31 - SPINE: Poller[1] PID[35241] PT[140065365190400] Device[294] HT[1] DS[858] TT[44.03] SNMP: v2: XX.XX.XX.XX, dsname: traffic_in, oid: .1.3.6.1.2.1.2.2.1.10.5, value: 2418660034
2023/09/07 16:19:31 - SPINE: Poller[1] PID[35241] PT[140065365190400] Device[294] HT[1] DS[858] TT[44.80] SNMP: v2: XX.XX.XX.XX, dsname: traffic_out, oid: .1.3.6.1.2.1.2.2.1.16.5, value: 1097624192
2023/09/07 16:19:31 - SPINE: Poller[1] PID[35241] PT[140065365190400] Device[294] HT[1] DS[859] TT[45.23] SNMP: v2: XX.XX.XX.XX, dsname: traffic_in, oid: .1.3.6.1.2.1.2.2.1.10.22, value: 1317513820
2023/09/07 16:19:31 - SPINE: Poller[1] PID[35241] PT[140065365190400] Device[294] HT[1] DS[859] TT[45.66] SNMP: v2: XX.XX.XX.XX, dsname: traffic_out, oid: .1.3.6.1.2.1.2.2.1.16.22, value: 3406978023
2023/09/07 16:19:31 - SPINE: Poller[1] PID[35241] PT[140065365190400] Device[294] HT[1] DS[860] TT[46.29] SNMP: v2: XX.XX.XX.XX, dsname: traffic_in, oid: .1.3.6.1.2.1.2.2.1.10.23, value: 51675649
2023/09/07 16:19:31 - SPINE: Poller[1] PID[35241] PT[140065365190400] Device[294] HT[1] DS[860] TT[46.84] SNMP: v2: XX.XX.XX.XX, dsname: traffic_out, oid: .1.3.6.1.2.1.2.2.1.16.23, value: 45007266
2023/09/07 16:19:31 - SPINE: Poller[1] PID[35241] PT[140065365190400] DEBUG: Setting up writes to local database
2023/09/07 16:19:31 - SPINE: Poller[1] PID[35241] PT[140065365190400] Device[294] HT[1] Total Time: 0.65 Seconds
2023/09/07 16:19:31 - SPINE: Poller[1] PID[35241] PT[140065365190400] DEBUG: Freeing Local Pool ID 1
2023/09/07 16:19:31 - SPINE: Poller[1] PID[35241] PT[140065365190400] DEBUG: Device[294] HT[1] DEBUG: HOST COMPLETE: About to Exit Device Polling Thread Function
2023/09/07 16:19:31 - SPINE: Poller[1] PID[35241] PT[140065406259392] The final count of Threads is 0
2023/09/07 16:19:31 - SPINE: Poller[1] PID[35241] PT[140065406259392] INFO: Thread complete for Device[0] and 0 to 0 sources
2023/09/07 16:19:31 - SPINE: Poller[1] PID[35241] PT[140065406259392] INFO: Thread complete for Device[294] and 0 to 0 sources
2023/09/07 16:19:31 - SPINE: Poller[1] PID[35241] PT[140065406259392] DEBUG: Closing Local Connection Pool ID 0
2023/09/07 16:19:31 - SPINE: Poller[1] PID[35241] PT[140065406259392] DEBUG: Closing Local Connection Pool ID 1
2023/09/07 16:19:31 - SPINE: Poller[1] PID[35241] PT[140065406259392] DEBUG: Closing Local Connection Pool ID 2
2023/09/07 16:19:31 - SPINE: Poller[1] PID[35241] PT[140065406259392] DEBUG: Closing Local Connection Pool ID 3
2023/09/07 16:19:31 - SPINE: Poller[1] PID[35241] PT[140065406259392] DEBUG: Thread Cleanup Complete
2023/09/07 16:19:31 - SPINE: Poller[1] PID[35241] PT[140065406259392] DEBUG: PHP Script Server Pipes Closed
2023/09/07 16:19:31 - SPINE: Poller[1] PID[35241] PT[140065406259392] DEBUG: Allocated Variable Memory Freed
2023/09/07 16:19:31 - SPINE: Poller[1] PID[35241] PT[140065406259392] DEBUG: MYSQL Free & Close Completed
2023/09/07 16:19:31 - SPINE: Poller[1] PID[35241] PT[140065406259392] DEBUG: Net-SNMP Close Completed
2023/09/07 16:19:31 - SPINE: Poller[1] PID[35241] PT[140065406259392] Time: 1.1636 s, Threads: 4, Devices: 2
```

snmp get/walk on one device:
```

# snmpget -v2c -c enomet -On XX.XX.XX.XX .1.3.6.1.6.3.10.2.1.3.0
.1.3.6.1.6.3.10.2.1.3.0 = No Such Instance currently exists at this OID
# snmpwalk -v2c -c enomet -On XX.XX.XX.XX .1.3.6.1.6.3.10.2.1.3.0
.1.3.6.1.6.3.10.2.1.3.0 = No Such Instance currently exists at this OID
```

**Server (please complete the following information):**
 - cacti: [1.2.25]
 - spine [1.2.25]

**Comment by TheWitness** [comment]
Do those errors appear at Normal level?  These really are not errors, just us informing the user at debug that the snmp agent does not support per second uptime.

**Comment by netniV** [comment]
I suspect you have your logging to high and agree with @TheWitness

**Comment by TheWitness** [comment]
Well, we should log that as a warning as it's not material.  Cacti is checking for the UPTIME OID, and if it does not find it, it finds the other one.  Yes, and the logging is HIGH.  I'll do a little fix up in a minute to avoid confusion in the future.

**Comment by TheWitness** [comment]
This is fixed in the 1.2.x branch now.

**Comment by netniV** [comment]
Was this applied to 1.3.x?
